standard/links_stream.lua

```diff
@@ -83,47 +86,135 @@ function StreamLinks.processStreams(forwardedInputArgs)
+	end
+
+	return platformStreams
+end
 
-			local key = StreamLinks.StreamKey(platformName):toString()
-			streams[key] = streamValue
-			streams[platformName] = streamValue -- Legacy
+---@param platform string
+---@param streamValue string
+---@return string?
+function StreamLinks.displaySingle(platform, streamValue)
+	local icon = '<i class="lp-icon lp-icon-21 lp-' .. (PLATFORM_TO_ICON[platform] or platform) .. '"></i>'
+	if platform == TL_STREAM then
+		return Page.makeExternalLink(icon, 'https://tl.net/video/streams/' .. streamValue)
+	end
+
+	local streamLink = StreamLinks.resolve(platform, streamValue)
+	if not streamLink then return nil end
+
+	return Page.makeInternalLink({}, icon, 'Special:Stream/' .. platform .. '/' .. streamValue)
+end
+
+---@param streams {string: string[]}
+---@param options {addSpace: boolean?}?
+---@return string?
+function StreamLinks.display(streams, options)
+	options = options or {}
+	local displays = {}
+	Array.forEach(StreamLinks.countdownPlatformNames, function(platform)
+		Array.forEach(streams[platform] or {}, function(streamValue)
+			table.insert(displays, StreamLinks.displaySingle(platform, streamValue))
+		end)
+	end)
+	if Table.isEmpty(displays) then return nil end
+	return table.concat(displays, options.addSpace and ' ' or nil)
+end
+
+---filters streams so that if english streams are present only those are returned
+---@param streamsInput {string: string}
+---@return {string: string[]}
+function StreamLinks.filterStreams(streamsInput)
+	local hasEnglishStream = Table.any(streamsInput, function(key)
+		return key:match('_en_') or Table.includes(StreamLinks.countdownPlatformNames, key)
+	end)
+
+	local streams = {}
+	for rawHost, stream in Table.iter.spairs(streamsInput) do
+		if #(mw.text.split(rawHost, '_', true)) == 3 then
+			local streamKey = StreamLinks.StreamKey(rawHost)
+			local platform = streamKey.platform
+			if not streams[platform] then
+				streams[platform] = {}
+			end
+			table.insert(streams[platform], (not hasEnglishStream or streamKey.languageCode == 'en') and stream or nil)
 		end
 	end
 
+	Array.forEach(StreamLinks.countdownPlatformNames, function(platformName)
+		local stream = streamsInput[platformName]
+		if type(streams[platformName]) == 'table' or String.isEmpty(stream) then return end
+		streams[platformName] = {stream, streamsInput[platformName .. 2]}
+	end)
+
 	return streams
 end
 
 --- StreamKey Class
 -- Contains the triplet that makes up a stream key
 -- [platform, languageCode, index]
----@class StreamKey
+---@class StreamKey: BaseClass
 ---@operator call(...): StreamKey
 ---@field platform string
 ---@field languageCode string
 ---@field index integer
----@field is_a function
-StreamLinks.StreamKey = Class.new(
+local StreamKey = Class.new(
 	function (self, ...)
 		self:new(...)
 	end
 )
-local StreamKey = StreamLinks.StreamKey
+StreamLinks.StreamKey = StreamKey
 
----@param tbl string
+---@param tbl string|StreamKey
```

Comment:
Shouldn't be there, there's an overload for that

Reply:
it throws anno warnings in that case:
![Screenshot 2024-07-08 123911](https://github.com/Liquipedia/Lua-Modules/assets/75081997/69f7b62f-b923-4505-b758-a3d77285d74e)
![Screenshot 2024-07-08 123948](https://github.com/Liquipedia/Lua-Modules/assets/75081997/266023dc-30f6-4c0a-b3bc-1064b3608d08)

Reply:
It's this open issue with the LSP
https://github.com/LuaLS/lua-language-server/issues/2708

Reply:
```suggestion
---@param tbl string
```